File: LearningAccumulation/app/src/main/java/com/xhj/learningaccumulation/MainActivity.java
package com.xhj.learningaccumulation;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import com.yanzhenjie.permission.Permission;

public class MainActivity extends BaseActivity  {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    @Override
    protected void initView() {
        TextView tv = (TextView) findViewById(R.id.tv);
        tv.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
//                getLocation();
                getCameraAndPhoto();
            }
        });
    }

    @Override
    protected void initData() {
        PermissionRequest permissionRequest = new PermissionRequest(this, new PermissionRequest.PermissionCallback() {
            @Override
            public void onSuccessful() {

            }

            @Override
            public void onFailure() {

            }
        });
        permissionRequest.request(Permission.STORAGE , Permission.CALENDAR);

    }
}

File: LearnKotlin/app/src/main/java/com/xuhaijia/learnkotlin/MainActivity.kt
package com.xuhaijia.learnkotlin

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*
import org.jetbrains.anko.toast

class MainActivity : AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        // java 8 下不支持 onClick 等方法
//        verticalLayout {
//            val name = editText()
//            button("Say Hello") {
//                setOnClickListener(View.OnClickListener {  toast("Hello, ${name.text}!")}) }
//
//            }
        setContentView(R.layout.activity_main)
        login.setOnClickListener({
            if ((account.text.toString().equals("1")) and (password.text.toString() == "2")) {
                toast("登录成功")
            } else {
                toast("登录失败" + account.text + "  : " + password.text)
            }
        })


    }
}
